Neos.Flow/Classes/ObjectManagement/Configuration/ConfigurationBuilder.php

```diff
@@ -561,7 +561,13 @@ protected function autowireProperties(array &$objectConfigurations)
                 if (!array_key_exists($propertyName, $properties)) {
                     /** @var Inject $injectAnnotation */
                     $injectAnnotation = $this->reflectionService->getPropertyAnnotation($className, $propertyName, Inject::class);
-                    $objectName = $injectAnnotation->name !== null ? $injectAnnotation->name : trim(implode('', $this->reflectionService->getPropertyTagValues($className, $propertyName, 'var')), ' \\');
+                    $objectName = $injectAnnotation->name;
+                    if ($objectName === null) {
+                        $objectName = $this->reflectionService->getPropertyType($className, $propertyName);
+                    }
+                    if ($objectName === null) {
+                        $objectName = trim(implode('', $this->reflectionService->getPropertyTagValues($className, $propertyName, 'var')), ' \\');
+                    }
                     $configurationProperty =  new ConfigurationProperty($propertyName, $objectName, ConfigurationProperty::PROPERTY_TYPES_OBJECT, null, $injectAnnotation->lazy);
```

Comment:
Can't we just decide here to ignore `$injectAnnotation->lazy` IF  we use `$this->reflectionService->getPropertyType` because in "legacy" code cases lazy should work fine?

Reply:
Okay, I gave it a try, see latest commit.